src/client/sound_openal.cpp

```diff
@@ -402,8 +403,30 @@ class OpenALSoundManager: public ISoundManager
 	{
 		std::unordered_map<std::string, std::vector<SoundBuffer*>>::iterator i =
 				m_buffers.find(name);
-		if(i == m_buffers.end())
-			return nullptr;
+		if (i == m_buffers.end()) {
+			auto filedatas = m_sound_files.find(name);
+			if (filedatas == m_sound_files.end())
+				return nullptr;
```

Comment:
`m_sound_files.erase(name);` missing?

Reply:
It's erased further down (line 421). The `return nullptr` here is for when there is no sound with the name in `m_sound_files` (and nothing to erase)